Libraries/LibWeb/HTML/HTMLButtonElement.cpp

```diff
@@ -29,6 +31,22 @@ void HTMLButtonElement::initialize(JS::Realm& realm)
     Base::initialize(realm);
 }
 
+void HTMLButtonElement::adjust_computed_style(CSS::ComputedProperties& style)
+{
+    // https://html.spec.whatwg.org/multipage/rendering.html#button-layout
+    // If the computed value of 'display' is 'inline-grid', 'grid', 'inline-flex', 'flex', 'none', or 'contents', then behave as the computed value.
```

Comment:
As discussed about the "behave as" thing: Once you've raised a spec issue, you don't have to wait for that to be resolved, but make sure to link to it here and note that it's `// AD-HOC` because different browsers do it differently.